### ISSUE TITLE
Canvas - race condition between "standalone" passenger (Nginx) and 'mod_passenger' (Apache)

### ISSUE THREAD

**Issue by JedMeister** [turnkeylinux/tracker#1495]
We're running Canvas via `mod_passenger` served by Apache. However, as [reported by Chris in the forums](https://www.turnkeylinux.org/forum/support/mon-20200727-0800/canvas-ami-installation) under some circumstance, Canvas is actually being served by Nginx!?

----

It took me a while to work out how or why, but deeper investigation reveals that despite the fact that we don't install Nginx, it's included with the source install of Passenger that we build (maybe we could remove that? Might slim the image down a little - but that's another issue for another day...). Regardless, we're providing a [`passenger`](https://github.com/turnkeylinux-apps/canvas/blob/master/overlay/etc/init.d/passenger) init.d script which runs Canvas via "standalone" passenger service.

So whichever service manages to lock ports 80 & 443 at boot, "wins" and will decide whether Canvas is running behind Nginx or Apache! Part of the reason why this hasn't become apparent sooner is because on face value, Canvas appears to work fine, regardless of which service is the one serving Canvas. The login page loads, the default user can log in and everything appears ok.

Another factor that has no doubt made discovery of this issue a little harder has been the fact that the systemd `passenger.service` (which systemd generates on the fly) reports "success" regardless of whether Nginix can lock ports 80 & 443.

From what I can gather, the only points where an issue becomes apparent is if you try to use Confconsole to generate an SSL cert or use the RCE API functionality.

Confconsole Let's Encrypt expects Nginix to be running via the `nginx.service`. As Nginx is locking port 80, Confconsole (or more correctly the Dehydrated wrapper) will try (and fail) to stop `nginx.service`. So when it's running via `passenger.service` the process will just fail (as per Chris' experience).

Re the RCE API, it's relatively new and is not required for all pages (and requires a valid SSL cert to work properly anyway). luck would have it that the times that I have tested it have coincidentally occurred when Apache was working for me as expected. However, as the PCE API is only configured to run as part of the Apache config, it won't work under the "standalone" Passenger/Nginx.

As noted in my most [recent post](https://www.turnkeylinux.org/forum/support/mon-20200727-0800/canvas-ami-installation#comment-48629) this should resolve the issue (BTW I've tested it myself and it appeared to fix the issue but I'm still waiting on third party confirmation):

----
## Workaround

Note: AWS Marketplace users need to prefix these commands with `sudo`.
```
systemctl stop passenger
systemctl restart apache2
```
Then check that everything appears to be ok via your web browser. Please note that the initial load may take a little while. Assuming Canvas is working as expected, then disable the `passenger.service`:
```
systemctl disable passenger
```

**Comment by JedMeister** [comment]
Closing as this was resolved in v16.1 release.
